nixos/modules/services/logging/packetbeat.nix

```diff
@@ -0,0 +1,185 @@
+            enabled: true
+          - type: amqp
+            ports: [5672]
+          - type: cassandra
+            ports: [9042]
+          - type: dhcpv4
+            ports: [67, 68]
+          - type: dns
+            ports: [53]
+          - type: http
+            ports: [80, 8080, 8000, 5000, 8002]
+          - type: memcache
+            ports: [11211]
+          - type: mysql
+            ports: [3306,3307]
+          - type: pgsql
+            ports: [5432]
+          - type: redis
+            ports: [6379]
+          - type: thrift
+            ports: [9090]
+          - type: mongodb
+            ports: [27017]
+          - type: nfs
+            ports: [2049]
+          - type: tls
+            ports:
+              - 443   # HTTPS
+              - 993   # IMAPS
+              - 995   # POP3S
+              - 5223  # XMPP over SSL
+              - 8443
+              - 8883  # Secure MQTT
+              - 9243  # Elasticsearch
+
+        '';
+        description = ''
+          Configuration of what protocols packetbeat should gather info about.
+          See <link xlink:href='https://www.elastic.co/guide/en/beats/packetbeat/current/configuration-protocols.html'/>
+          for the configuration options available.
+        '';
+      };
+
+      extraConfig = mkOption {
+        type = types.lines;
+        default = ''
+          packetbeat.interfaces.device: any
+
+          setup.template.settings:
+            index.number_of_shards: 1
+
+          setup.kibana:
+            host: "localhost:5601"
+
+          output.elasticsearch:
+            hosts: ["localhost:9200"]
+
+          processors:
+            - # Add forwarded to tags when processing data from a network tap or mirror.
+              if.contains.tags: forwarded
+              then:
+                - drop_fields:
+                    fields: [host]
+              else:
+                - add_host_metadata: ~
+            - add_cloud_metadata: ~
+            - add_docker_metadata: ~
+        '';
+        description = "Any other configuration options you want to add";
+      };
+
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    assertions = [
+      {
+        assertion = !hasPrefix "/" cfg.stateDir;
+        message =
+          "The option services.packetbeat.stateDir shouldn't be an absolute directory." +
+          " It should be a directory relative to /var/lib/.";
+      }
+      {
+        assertion = cfg.configProtocols != "" || cfg.configFlows != "";
+        message =
+          "The options services.packetbeat.configProtocols and/or services.packetbeat.configFlows should" +
+          " be set or else packetbeat won't do anything useful and error out.";
+      }
+    ];
+
+    systemd.services.packetbeat = {
+      description = "Packetbeat log shipper";
+      wantedBy = [ "multi-user.target" ];
+      preStart = ''
+        mkdir -p ${cfg.stateDir}/data
+        mkdir -p ${cfg.stateDir}/logs
+      '';
+      serviceConfig = {
```

Comment:
This is completely copied from the official unit, packetbeat needs to run as root because it captures packets from interfaces. I dunno if CAP_NET_ADMIN might be enough instead of having root, but this is how the official systemd unit does it.